src/main/webapp/components/utils/getField.js

```diff
@@ -7,7 +7,9 @@
  * @param defaultValue the value to be used if obj[field] === undefined.
```

Comment:
not related to this CL, but generally speaking accessing properties off of objects using `[]` syntax is an anti-pattern.
* If your code needs to map keys to values, consider using a proper ES6 `Map`.
* If your properties are well known in advance, it's usually not a good idea to attempt to reflect over them: prefer accessing them using regular `.foo`.
* If your properties are potentially `null` or `undefined`
  * the first thing to do is avoid having the problem in the first place. Make sure that all values are set once a value enters the application (e.g. as you parse JSON), and later on always crash the application if a value that should be there isn't.
  * if your application has a value that can (by app semantics) truly not exist, then using `null` or `undefined` is fine. In that case, you can use `??` or `?.` to access it in a reasonably convenient fashion.
  * do not use `?.` syntax to allow null values to flow through your application. It's relatively easy to debug a null crash at the first place where a value appears. It's very hard to debug such a crash several `?.` accesses later.

HTH.

Reply:
Very helpful! I'm considering deprecating <code>getField</code> and <code>getFields</code>. 

Indeed, our app is built so that such values cannot be null / undefined from the way our API serves the data, so perhaps using these methods in this way is redundant.

The original utility of <code>getField</code> was to be used in conjunction with <code>getFields</code> to return a subset of fields from an object. That being said, this can be achieved with **destructing** instead.

> the first thing to do is avoid having the problem in the first place. Make sure that all values are set once a value enters the application (e.g. as you parse JSON), and later on always crash the application if a value that should be there isn't.

Yes I see how letting the app crash in these contexts is very useful for debugging!

Thanks for the insight.